Portal/Controllers/WasteController.cs

```diff
@@ -34,7 +34,7 @@ public async Task<IActionResult> WhatHaveYouDoneWaste(int id, WhatHaveYouDoneWas
                 return View(whatHaveYouDoneWaste);
             }
 
-            await _wasteService.SaveSelectedWasteType(whatHaveYouDoneWaste.JourneyId, whatHaveYouDoneWaste.SelectedWaste);
+            await _wasteService.SaveWhatHaveYouDoneWaste(whatHaveYouDoneWaste.JourneyId, whatHaveYouDoneWaste.SelectedWaste);
```

Comment:
I would suggest not passing the individual parameters, but the entire object. That was the service can perform any checks against the object instead of relying on it being populated at the controller level

Reply:
Done.